include/zenohcxx/api.hxx

```diff
@@ -670,6 +670,93 @@ struct Timestamp : Copyable<::z_timestamp_t> {
+struct AttachmentView : public Copyable<::z_attachment_t> {
+    using Copyable::Copyable;
+
+    /// @name Constructors
+
+    /// @brief AttachmentView construnctor
+    AttachmentView(const void* _data, z_attachment_iter_driver_t _iter_driver) : Copyable({_data, _iter_driver}) {}
+
+    /// @name Methods
+
+    /// @brief Set the data pointer for attachment
+    /// @param _data value of ``void*`` type
+    /// @return Reference to the ``AttachmentView`` object
+    AttachmentView& set_data(const void* _data) {
+        data = _data;
+        return *this;
+    }
+
+    /// @brief Set the iteration driver
+    /// @param _iter_driver value of ``z_attachment_iter_driver_t`` type
+    /// @return Reference to the ``AttachmentView`` object
+    AttachmentView& set_iteration_driver(z_attachment_iter_driver_t _iter_driver) {
+        iteration_driver = _iter_driver;
+        return *this;
+    }
+
+    /// @brief Get the data pointer of the attachment
+    /// @return value of ``void *`` type
+    const void* get_data() const { return data; }
+
+    /// @brief Get the iteration driver
+    /// @return value of ``z_attachment_iter_driver_t`` type
+    z_attachment_iter_driver_t get_iteration_driver() const { return iteration_driver; }
+
+    /// @name Methods
+
+    /// Returns the item value from the attachment by key
+    /// @return the item value
+    BytesView get(const BytesView& key) const { return ::z_attachment_get(*this, key); }
+
+    /// Checks if the attachment is initialized
+    /// @return true if the attachment is initialized
+    bool check() const { return ::z_attachment_check(this); }
+
+    typedef int8_t (*IterBody)(const BytesView& key, const BytesView& value, void* context);
+
+    // Iterate over attachment's key-value pairs, breaking if `body` returns a non-zero
+    // value for a key-value pair, and returning the latest return value.
+    // `context` is passed to `body` to allow stateful closures.
+    // This function takes no ownership whatsoever.
+    /// @return the latest return value
+    int8_t iterate(IterBody body, void* context) const;
+
+    /// @name Operators
+
+    /// @brief Equality operator
+    /// @param v other ``AttachmentView`` object
+    /// @return true if the attachment objects encodings are equal
+    bool operator==(const AttachmentView& v) const {
+        return get_data() == v.get_data() && get_iteration_driver() == v.get_iteration_driver();
+    }
+
+    /// @brief Inequality operator
+    /// @param v other ``AttachmentView`` object
+    /// @return true if the attachment objects are not equal
+    bool operator!=(const AttachmentView& v) const { return !operator==(v); }
+};
+
+/// Wraps the container which allows iterate by std::pair<std::string_view, std::string_view>
+// (e.g. std::map<std::string_view, std::string_view>) as an attachment.
+template <typename T>
+inline AttachmentView as_attachment(const T& pair_container) {
```

Comment:
In other "view" classes similar functionality is implemented in constructor (https://github.com/eclipse-zenoh/zenoh-cpp/blob/main/include/zenohcxx/api.hxx#L289). Is there any reason to implement it as a separate function here?

Reply:
Sound good, thanks!

Reply:
Done!